### PR TITLE
Initial work on adding iCasework case management

### PR DESCRIPTION
**Pull request by gbp**

### CODE REVIEW

File: app/models/case_management/icasework.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      @client = client || ::Icasework::Case
+    end
+
+    def submit_foi_request!(name:, email:, body:)
+      request = client.create(
+        request_method: 'Online Form',
+        type: 'InformationRequest',
+        name: name,
+        email: email,
+        details: body
```
Comment (garethrees) [inline app/models/case_management/icasework.rb:16]: The raw post data should be:

```
     -d '{"Type":"InformationRequest",
           "RequestMethod":"Online Form",
           "Customer.Name":"Alice Brown",
           "Customer.Email":"alice@example.com",
           "Details":"test foi"}'
```

Do we convert these keys correctly. Looks like there [might be some todos](https://github.com/mysociety/icasework-ruby/blob/48a666bcae39ef3f1825027c7b4cd940450ad890/lib/icasework/resource.rb#L123-L126) for this to work as is?. I'd be inclined to use string keys so that we don't have to jump through hoops to translate between the "official" API and our client library. The point of this `CaseManagement` interface is that we only have to deal with the grossness in one place?

```ruby
client.create(
  'RequestMethod' => 'Online Form',
  'Type' => 'InformationRequest',
  'Customer.Name' => name,
  'Customer.Email' => email,
  'Details' => body
)
```

File: app/models/case_management/icasework.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      )
+
+      SubmittedRequest.new(request)
+    end
+
+    def published_requests(query_params)
```
Comment (garethrees) [inline app/models/case_management/icasework.rb:22]: Since this doesn't "work" as is (because of API issues) let's extract this part out into separate commits so that we can merge the submission of requests as soon as its working.

File: app/models/case_management/icasework.rb (hunk 1 above)
Comment (gbp) [reply app/models/case_management/icasework.rb:16]: From my tests the endpoint does accept `Name` and `Email`. Not saying to should uses these as is seems undocumented. My intention and some of the work in progress I've done on the gem would handle this nicely EG:
```ruby
client.create(
  request_method: 'Online Form',
  type: 'InformationRequest',
  customer: {
    name: name,
    email: email,
  },
  details: body
)
```

Comment (garethrees) [reply app/models/case_management/icasework.rb:16]: > From my tests the endpoint does accept `Name` and `Email`

Worth posting a BC question about this and whether there's any difference.

**Comment by garethrees** [comment]
Closed in favour of https://github.com/mysociety/foi-for-councils/pull/400 and https://github.com/mysociety/foi-for-councils/pull/401.
